### PR TITLE
Node: Reduce auto reobservation requests

### PR DESCRIPTION
**Pull request by bruce-riley**

### CODE REVIEW

**Review by panoel — DISMISSED** [review]

**Review by tbjump — APPROVED** [review]
Here is a PR that rebases this change on the test improvements: https://github.com/wormhole-foundation/wormhole/pull/3205 
The unit tests there are passing and the code coverage is good, so I'm fine with merging this in to main.

**Comment by tbjump** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3203** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/3203" target="_blank">Graphite</a>  👈
    * **PR #3207** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/3207" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/3203?utm_source=stack-comment).

**Review by panoel — APPROVED** [review]

File: node/pkg/processor/cleanup.go [hunk 1]
```diff
@@ -203,3 +216,38 @@ func (p *Processor) handleCleanup(ctx context.Context) {
...
+	}
+
+	v, err := vaa.Unmarshal(vb)
+	if err != nil {
+		return false, fmt.Errorf("failed to unmarshal VAA: %w", err)
+	}
+
+	oldHash := hex.EncodeToString(v.SigningDigest().Bytes())
+	if hash != oldHash {
+		p.logger.Debug("VAA already in DB but hash is different", zap.String("old_hash", oldHash), zap.String("new_hash", hash))
```
Comment (evan-gray) [inline node/pkg/processor/cleanup.go:249]: is this missing a `return false, nil`

**Review by evan-gray — CHANGES_REQUESTED** [review]

**Review by tbjump — APPROVED** [review]

File: node/pkg/processor/cleanup.go (hunk 1 above)
Comment (bruce-riley) [reply node/pkg/processor/cleanup.go:249]: Apparently @tbjump fixed this.

**Review by panoel — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]

**Pull request merged**